js/core/utils/common.js

```diff
@@ -268,10 +266,12 @@ var equalByValue = function(object1, object2, deep) {
     return false;
 };
 
-var getKeyHash = function(key) {
-    if(typeUtils.isObject(key) || Array.isArray(key)) {
+const getKeyHash = function(key) {
+    if(key instanceof Guid) {
```

Comment:
Can we use valueOf here?
```
    if (isFunction(key.valueOf)) {
      key= key.valueOf();
    }
```

Reply:
`Object` and `Array` has the `valueOf` method too at it returns an object. It can break our selection logic